File: Python/Calculator.py
import math
import cmath

def basic_operations(a, b, operation):
    if operation == '+':
        return a + b
    elif operation == '-':
        return a - b
    elif operation == '*':
        return a * b
    elif operation == '/':
        return a / b if b != 0 else "Error: Division by zero"
    else:
        return "Invalid Operation"

def advanced_operations(a, operation):
    if operation == 'sqrt':
        return math.sqrt(a)
    elif operation == 'log':
        return math.log(a)
    elif operation == 'exp':
        return math.exp(a)
    elif operation == 'sin':
        return math.sin(a)
    elif operation == 'cos':
        return math.cos(a)
    elif operation == 'tan':
        return math.tan(a)
    elif operation == 'complex_sqrt':
        return cmath.sqrt(a)
    else:
        return "Invalid Advanced Operation"

def complex_calculator():
    while True:
        print("\nChoose mode: Basic or Advanced or Complex?")
        mode = input("Mode (basic/advanced/complex): ").lower()

        if mode == 'basic':
            a = float(input("Enter first number: "))
            operation = input("Enter operation (+, -, *, /): ")
            b = float(input("Enter second number: "))
            result = basic_operations(a, b, operation)
            print(f"Result: {result}")

        elif mode == 'advanced':
            a = float(input("Enter number: "))
            operation = input("Enter operation (sqrt, log, exp, sin, cos, tan): ").lower()
            result = advanced_operations(a, operation)
            print(f"Result: {result}")

        elif mode == 'complex':
            a = complex(input("Enter complex number: "))
            operation = input("Enter operation (complex_sqrt): ").lower()
            result = advanced_operations(a, operation)
            print(f"Result: {result}")

        else:
            print("Invalid mode!")
        
        continue_calculation = input("Do you want to perform another calculation? (y/n): ").lower()
        if continue_calculation != 'y':
            break

if __name__ == "__main__":
    complex_calculator()

File: C++/Calculator.cpp
#include <iostream>
#include <cmath>

using namespace std;

double basic_operations(double a, double b, char operation) {
    switch (operation) {
        case '+': return a + b;
        case '-': return a - b;
        case '*': return a * b;
        case '/': return (b != 0) ? a / b : NAN;
        default: return NAN;
    }
}

double advanced_operations(double a, string operation) {
    if (operation == "sqrt") return sqrt(a);
    else if (operation == "log") return log(a);
    else if (operation == "exp") return exp(a);
    else if (operation == "sin") return sin(a);
    else if (operation == "cos") return cos(a);
    else if (operation == "tan") return tan(a);
    else return NAN;
}

int main() {
    double a, b;
    char operation;
    string mode, adv_operation;

    cout << "Choose mode: basic or advanced?\n";
    cin >> mode;

    if (mode == "basic") {
        cout << "Enter first number: ";
        cin >> a;
        cout << "Enter operation (+, -, *, /): ";
        cin >> operation;
        cout << "Enter second number: ";
        cin >> b;
        cout << "Result: " << basic_operations(a, b, operation) << endl;
    } else if (mode == "advanced") {
        cout << "Enter number: ";
        cin >> a;
        cout << "Enter operation (sqrt, log, exp, sin, cos, tan): ";
        cin >> adv_operation;
        cout << "Result: " << advanced_operations(a, adv_operation) << endl;
    } else {
        cout << "Invalid mode!";
    }

    return 0;
}
